File: src/main/resources/archetype-resources/src/main/java/sample/service/UserService.java
#set( $symbol_pound = '#' )
#set( $symbol_dollar = '$' )
#set( $symbol_escape = '\' )
package ${package}.sample.service;

import ${package}.sample.domain.User;
import ${package}.sample.domain.dto.UserDTO;
import ${package}.sample.service.generic.PrivateService;
import org.springframework.stereotype.Service;

/**
 * @author Vittorio Valent
 *
 */
@Service
public class UserService extends PrivateService<User, UserDTO> {

}

File: src/main/resources/archetype-resources/frontend/src/service/user.service.ts
import { Injectable } from '@angular/core';
import { AbstractService } from './abstractservice';
import { UserDTO } from '../dto/user-dto';
import { HttpClient } from '@angular/common/http';
import { LoginDTO } from '../dto/login-dto';
import { Observable } from 'rxjs';
import { AuthToken } from '../dto/auth-token';
/**
 * I service sono decorati da @Injectable.
 * Qui trovate, oltre ai metodi ereditati dall'Abstract,
 *  il metodo per il login (in mirror con il backend).
 *
 * @author Vittorio Valent
 *
 * @see AbstractService
 */
@Injectable({
  providedIn: 'root'
})
export class UserService extends AbstractService<UserDTO> {
  constructor(http: HttpClient) {
    super(http);
    this.type = 'user';
  }
  login(loginDTO: LoginDTO): Observable<AuthToken> {
    return this._http.post<AuthToken>('http://localhost:8081/authentication/public/authenticate', loginDTO);
  }
}
